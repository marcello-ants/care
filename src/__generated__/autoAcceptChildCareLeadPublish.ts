/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AutoAcceptChildCareLeadPublishInput } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: autoAcceptChildCareLeadPublish
// ====================================================

export interface autoAcceptChildCareLeadPublish_autoAcceptChildCareLeadPublish {
  __typename: "AutoAcceptChildCareLeadPublishResponse";
  /**
   * Batch id of creation lead operation
   */
  batchId: string | null;
}

export interface autoAcceptChildCareLeadPublish {
  /**
   * Creates an auto-accept lead
   */
  autoAcceptChildCareLeadPublish: autoAcceptChildCareLeadPublish_autoAcceptChildCareLeadPublish;
}

export interface autoAcceptChildCareLeadPublishVariables {
  autoAcceptChildCareLeadPublishInput: AutoAcceptChildCareLeadPublishInput;
}
