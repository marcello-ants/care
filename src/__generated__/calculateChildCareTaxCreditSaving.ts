/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { ChildCareTaxCreditSavingCalculationInput, SubServiceType } from "./globalTypes";

// ====================================================
// GraphQL query operation: calculateChildCareTaxCreditSaving
// ====================================================

export interface calculateChildCareTaxCreditSaving_calculateChildCareTaxCreditSaving_ChildCareTaxCreditSavingCalculationSuccess_options_originalTotal {
  __typename: "Money";
  /**
   * Amount of money
   */
  amount: globalScalarDecimal;
  /**
   * Type of currency
   */
  currencyCode: globalScalarCurrency;
}

export interface calculateChildCareTaxCreditSaving_calculateChildCareTaxCreditSaving_ChildCareTaxCreditSavingCalculationSuccess_options_discountedTotal {
  __typename: "Money";
  /**
   * Amount of money
   */
  amount: globalScalarDecimal;
  /**
   * Type of currency
   */
  currencyCode: globalScalarCurrency;
}

export interface calculateChildCareTaxCreditSaving_calculateChildCareTaxCreditSaving_ChildCareTaxCreditSavingCalculationSuccess_options_savingAmount {
  __typename: "Money";
  /**
   * Amount of money
   */
  amount: globalScalarDecimal;
  /**
   * Type of currency
   */
  currencyCode: globalScalarCurrency;
}

export interface calculateChildCareTaxCreditSaving_calculateChildCareTaxCreditSaving_ChildCareTaxCreditSavingCalculationSuccess_options {
  __typename: "TaxCreditSaving";
  /**
   * Total amount for child care subservice without tax credit
   */
  originalTotal: calculateChildCareTaxCreditSaving_calculateChildCareTaxCreditSaving_ChildCareTaxCreditSavingCalculationSuccess_options_originalTotal;
  /**
   * Total amount for child care subservice including tax credit discount
   */
  discountedTotal: calculateChildCareTaxCreditSaving_calculateChildCareTaxCreditSaving_ChildCareTaxCreditSavingCalculationSuccess_options_discountedTotal;
  /**
   * Percentage of saving that Seeker can get on child care subservice due to tax credit
   */
  savingPercentage: number;
  /**
   * Saving amount that Seeker can get on child care subservice due to tax credit
   */
  savingAmount: calculateChildCareTaxCreditSaving_calculateChildCareTaxCreditSaving_ChildCareTaxCreditSavingCalculationSuccess_options_savingAmount;
  /**
   * Subservice type
   */
  subServiceType: SubServiceType;
}

export interface calculateChildCareTaxCreditSaving_calculateChildCareTaxCreditSaving_ChildCareTaxCreditSavingCalculationSuccess {
  __typename: "ChildCareTaxCreditSavingCalculationSuccess";
  /**
   * List of Child Care subservices with possible tax credit saving info
   */
  options: calculateChildCareTaxCreditSaving_calculateChildCareTaxCreditSaving_ChildCareTaxCreditSavingCalculationSuccess_options[];
}

export interface calculateChildCareTaxCreditSaving_calculateChildCareTaxCreditSaving_ChildCareTaxCreditSavingCalculationError {
  __typename: "ChildCareTaxCreditSavingCalculationError";
  /**
   * Summary of the error.
   */
  message: string | null;
}

export type calculateChildCareTaxCreditSaving_calculateChildCareTaxCreditSaving = calculateChildCareTaxCreditSaving_calculateChildCareTaxCreditSaving_ChildCareTaxCreditSavingCalculationSuccess | calculateChildCareTaxCreditSaving_calculateChildCareTaxCreditSaving_ChildCareTaxCreditSavingCalculationError;

export interface calculateChildCareTaxCreditSaving {
  /**
   * Calculate possible tax credit savings for Child Care subservices based on area, number of children and required number of hours per week
   */
  calculateChildCareTaxCreditSaving: calculateChildCareTaxCreditSaving_calculateChildCareTaxCreditSaving;
}

export interface calculateChildCareTaxCreditSavingVariables {
  input: ChildCareTaxCreditSavingCalculationInput;
}
