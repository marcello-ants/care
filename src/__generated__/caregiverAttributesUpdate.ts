/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { CaregiverAttributesUpdateInput } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: caregiverAttributesUpdate
// ====================================================

export interface caregiverAttributesUpdate_caregiverAttributesUpdate {
  __typename: "CaregiverAttributesUpdateSuccess";
  /**
   * Dummy field
   */
  dummy: string | null;
}

export interface caregiverAttributesUpdate {
  /**
   * # Updates caregiver attributes
   */
  caregiverAttributesUpdate: caregiverAttributesUpdate_caregiverAttributesUpdate;
}

export interface caregiverAttributesUpdateVariables {
  input: CaregiverAttributesUpdateInput;
}
