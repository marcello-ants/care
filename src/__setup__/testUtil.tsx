/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { NextRouter } from 'next/router';
import { render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { RouterContext } from 'next/dist/shared/lib/router-context';
import DayJSUtils from '@date-io/dayjs';
import { theme } from '@care/material-ui-theme';
import { ThemeProvider } from '@material-ui/core';
import { FeatureFlags, FeatureFlagsProvider } from '@/components/FeatureFlagsContext';
import { AppStateProvider } from '@/components/AppState';
import { initialAppState } from '@/state';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { AppState } from '@/types/app';

export interface PrepareTreeOptions {
  pathname?: string;
  mocks?: MockedResponse[];
  flags?: FeatureFlags;
  appState?: AppState;
  addTypeName?: boolean;
  basePath?: string;
}

const prepareTree = (options?: PrepareTreeOptions) => {
  const {
    pathname = '/path',
    mocks = [],
    flags = {},
    appState = initialAppState,
    addTypeName = false,
    basePath = '/',
  } = options || {};
  const routerMock = {
    push: jest.fn(),
    asPath: pathname,
    beforePopState: (cb: any) => {
      cb({ url: pathname });
    },
    basePath,
    pathname,
    query: {},
    back: jest.fn(),
  } as unknown as NextRouter;

  const renderFn = (children: React.ReactNode) => (
    <ThemeProvider theme={theme}>
      <RouterContext.Provider value={routerMock}>
        <MockedProvider mocks={mocks} addTypename={addTypeName}>
          <MuiPickersUtilsProvider utils={DayJSUtils}>
            <FeatureFlagsProvider flags={flags}>
              <AppStateProvider initialStateOverride={appState}>{children}</AppStateProvider>
            </FeatureFlagsProvider>
          </MuiPickersUtilsProvider>
        </MockedProvider>
      </RouterContext.Provider>
    </ThemeProvider>
  );

  return {
    renderFn,
    routerMock,
  };
};

export const preRenderHook = (options?: PrepareTreeOptions) => {
  const tree = prepareTree(options);
  const renderFn = <T,>(hook: () => T) =>
    renderHook(hook, {
      wrapper({ children }: { children: React.ReactNode }) {
        return tree.renderFn(children);
      },
    });

  return {
    ...tree,
    renderFn,
  };
};

export const preRenderPage = (options?: PrepareTreeOptions) => {
  const tree = prepareTree(options);
  const renderFn = (ComponentOrChildren: React.ComponentType | React.ReactNode) => {
    if (React.isValidElement(ComponentOrChildren)) {
      return render(tree.renderFn(ComponentOrChildren));
    }

    const Component = ComponentOrChildren as React.ComponentType;
    return render(tree.renderFn(<Component />));
  };

  return {
    ...tree,
    renderFn,
    renderTree: tree.renderFn,
  };
};

export const setupWindowLocation = () => {
  const originalLocation = window.location;
  const windowLocationMock = jest.fn();

  // @ts-ignore
  delete window.location;
  // @ts-ignore
  window.location = {
    assign: windowLocationMock,
  };

  return {
    mock: windowLocationMock,
    cleanUp() {
      window.location = originalLocation;
    },
  };
};

export const setupLocalStorage = () => {
  const originalLocalStorage: Storage = window.localStorage;
  const localStorageMock = jest.fn();
  Object.defineProperty(window, 'localStorage', localStorageMock);

  return {
    mock: localStorageMock,
    cleanUp() {
      Object.defineProperty(window, 'localStorage', originalLocalStorage);
    },
  };
};

export function mockFetch({ ok, status, data }: { ok: boolean; status: number; data: any }) {
  return () =>
    Promise.resolve({
      ok,
      status,
      json: () => Promise.resolve(data),
    } as Response);
}
