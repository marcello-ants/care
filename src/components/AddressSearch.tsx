import { useEffect, useState } from 'react';
import { theme } from '@care/material-ui-theme';
import { Box, Button, makeStyles } from '@material-ui/core';
import { GoogleMaps, GoogleMapsActions, useGoogleMaps } from '@care/seeker-react-components';
import { Icon24UtilitySearch } from '@care/react-icons';
import { InlineTextField, Typography } from '@care/react-component-lib';
import { AddressInput } from '@/__generated__/globalTypes';
import { logIbAddressErrorEvent } from '@/utilities/analyticsPagesUtils';

const useStyles = makeStyles(() => ({
  addressSearchContainer: {
    position: 'relative',
    marginTop: theme.spacing(-2.75),
    '&& > .MuiTextField-root': {
      paddingLeft: 0,
      paddingRight: 0,
    },
    '&& label': {
      paddingLeft: theme.spacing(5),
      textAlign: 'left',
    },
    '&& input': {
      paddingLeft: theme.spacing(7),
    },
  },
  loopIcon: {
    position: 'absolute',
    top: 42,
    left: 20,
    zIndex: 2,
  },
  locationContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  locationText: {
    textAlign: 'left',
    whiteSpace: 'pre-wrap',
  },
  editLocationButton: {
    width: 81,
  },
  notFoundContainer: {
    padding: theme.spacing(5, 0),
    textAlign: 'center',
  },
  suiteContainer: {
    '&& > .MuiTextField-root': {
      paddingLeft: 0,
      paddingRight: 0,
    },
  },
}));

interface AddressSearchProps {
  seekerAddress: AddressInput;
  setSeekerAddress: (address: AddressInput) => void;
  displayAddress: string;
  setDisplayAddress: (displayAddress: string) => void;
  pageName: string;
}

const AddressSearch = ({
  seekerAddress,
  setSeekerAddress,
  displayAddress,
  setDisplayAddress,
  pageName,
}: AddressSearchProps): JSX.Element => {
  const classes = useStyles();
  const [{ placeDetails, predictions }, dispatch] = useGoogleMaps();
  const addressLabelText = 'Street address';
  const addressLabelTextOnInput = 'Search address';
  const suiteLabelText = 'Apt or Suite (optional)';
  const invalidAddressText = 'Invalid address';
  const notFoundText = 'Address not found';
  const notFoundDescription = `Make sure it's spelled correctly or try adding a city, state or zip code`;
  const CITY_POSITION: number = 2;
  const STATE_POSITION: number = 3;

  const [label, setLabel] = useState(addressLabelText);
  const [showAddressSearch, setShowAddressSearch] = useState<boolean>(true);
  const [shouldSaveAddress, setShouldSaveAddress] = useState<boolean>(false);

  const isValidAddress = seekerAddress.zip && seekerAddress.addressLine1;

  const handleFocus = () => {
    setLabel(addressLabelTextOnInput);
  };

  const handleBlur = () => {
    setLabel(addressLabelText);
  };

  const setGoogleMapsPlaceInfo = (inputText: string) => {
    dispatch({ type: GoogleMapsActions.SET_INPUT_SEARCH, payload: inputText });
    dispatch({ type: GoogleMapsActions.SET_PLACE_DETAILS, payload: null });
    dispatch({ type: GoogleMapsActions.CLEAR_PREDICTIONS });
  };

  const handleEditClick = () => {
    setGoogleMapsPlaceInfo(seekerAddress.addressLine1);
    setShowAddressSearch(true);
    setShouldSaveAddress(false);
  };

  const handlePredictionClick = () => {
    setShowAddressSearch(false);
    setShouldSaveAddress(true);
  };

  useEffect(() => {
    if (shouldSaveAddress && placeDetails) {
      setSeekerAddress({
        ...seekerAddress,
        addressLine1: placeDetails.prediction.structured_formatting.main_text,
        city: placeDetails.prediction.terms[CITY_POSITION].value,
        state: placeDetails.prediction.terms[STATE_POSITION].value,
        zip: placeDetails.zip || '',
        longitude: placeDetails.longitude,
        latitude: placeDetails.latitude,
      });
      setDisplayAddress(
        `${placeDetails.prediction.structured_formatting.main_text} \n${
          placeDetails.prediction.structured_formatting.secondary_text
        } ${placeDetails.zip || ''}`
      );
      setShouldSaveAddress(false);
    }
  }, [shouldSaveAddress, placeDetails]);

  useEffect(() => {
    if (displayAddress) {
      setShowAddressSearch(false);
    }
  }, []);

  const handleSuiteChange = (value: string) => {
    setSeekerAddress({
      ...seekerAddress,
      addressLine2: value,
    });
  };

  const addressSearchMode = showAddressSearch || shouldSaveAddress;

  useEffect(() => {
    if (addressSearchMode && !predictions) {
      logIbAddressErrorEvent(notFoundText, pageName);
    }
  }, [addressSearchMode, predictions]);
  useEffect(() => {
    if (!addressSearchMode && !isValidAddress) {
      logIbAddressErrorEvent(invalidAddressText, pageName);
    }
  }, [addressSearchMode, isValidAddress]);

  return (
    <Box>
      {addressSearchMode ? (
        <Box className={classes.addressSearchContainer}>
          <Icon24UtilitySearch className={classes.loopIcon} />
          <GoogleMaps.InputSearch label={label} onFocus={handleFocus} onBlur={handleBlur} />
          {predictions ? (
            predictions.map((prediction) => (
              <GoogleMaps.Prediction
                {...prediction}
                key={prediction.place_id}
                onClick={handlePredictionClick}
              />
            ))
          ) : (
            <Box className={classes.notFoundContainer}>
              <Typography variant="h2">{notFoundText}</Typography>
              <Typography variant="body1">{notFoundDescription}</Typography>
            </Box>
          )}
        </Box>
      ) : (
        <Box>
          <Box className={classes.locationContainer}>
            <Box className={classes.locationText}>
              <Typography>{displayAddress}</Typography>
            </Box>
            <Box>
              <Button
                color="secondary"
                variant="outlined"
                size="small"
                className={classes.editLocationButton}
                onClick={handleEditClick}>
                Edit
              </Button>
            </Box>
          </Box>
          {!isValidAddress && (
            <Box mt={2}>
              <Typography variant="body2" color="error">
                {invalidAddressText}
              </Typography>
            </Box>
          )}
          <Box className={classes.suiteContainer}>
            <InlineTextField
              id="aptOrSuite"
              InputLabelProps={{ shrink: false }}
              name="aptOrSuite"
              label={suiteLabelText}
              value={seekerAddress.addressLine2}
              onChange={(e) => handleSuiteChange(e.target.value)}
            />
          </Box>
        </Box>
      )}
    </Box>
  );
};

const AddressSearchWrapper = (props: AddressSearchProps) => {
  return (
    <GoogleMaps.Provider>
      <AddressSearch {...props} />
    </GoogleMaps.Provider>
  );
};

export default AddressSearchWrapper;
