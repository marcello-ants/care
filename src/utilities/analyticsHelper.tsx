/* istanbul ignore next */
import {
  Config,
  EventPayload,
  logEvent as _logEvent,
  init,
  setUserProps,
  getInstance,
} from '@care/analytics';
import dayjs from 'dayjs';
import getConfig from 'next/config';
import logger from '@/lib/clientLogger';
import { CLIENT_SIDE_ERROR_TAG } from '@/constants';
import { captureException, configureScope } from '@sentry/nextjs';
import { FeatureFlags } from '@/components/FeatureFlagsContext';
import { getCzenVisitorId, getCzenSessionId } from './czenCookiesHelper';
import getVertical from './verticalUtils';

const { publicRuntimeConfig } = getConfig();

export const AMPLITUDE_DATE_FORMAT = 'MMDDYYYY';
export const AMPLITUDE_TIME_FORMAT = 'hh:mm A';

let initialized = false;

/* istanbul ignore next */
function getCommonData() {
  const commonData: any = {
    job_flow: 'MW VHP enrollment',
    web_platform: window.innerWidth < 600 ? 'Mobile Web' : 'Desktop',
  };

  const czenSessionId = getCzenSessionId();
  if (czenSessionId) {
    commonData.czen_session_id = czenSessionId;
  }

  const czenVisitorId = getCzenVisitorId();
  if (czenVisitorId) {
    commonData.visitor_id = czenVisitorId;
  }

  return commonData;
}

const initialize = () => {
  if (!initialized) {
    const careAnalyticsConfig: Config = {
      analyticsEnabled: publicRuntimeConfig.ANALYTICS_ENABLED === 'true',
      key: publicRuntimeConfig.AMPLITUDE_API_KEY,
      commonData: getCommonData(),
    };
    init(careAnalyticsConfig);
    initialized = true;

    const amplitudeSessionId = getInstance()?.getSessionId();
    if (amplitudeSessionId) {
      configureScope((scope) => {
        const user = scope.getUser();
        scope.setUser({
          ...user,
          amplitudeSessionId,
        });
      });

      if (window.SplunkRum) {
        window.SplunkRum.setGlobalAttributes({
          'enduser.amplitudeSessionId': amplitudeSessionId,
        });
      }
    }
  }
};

interface AnalyticsHelperPayload extends EventPayload {
  vertical?: string;
}

export namespace AnalyticsHelper {
  /* istanbul ignore next */
  export const logEvent = (payload: AnalyticsHelperPayload) => {
    try {
      initialize();
      _logEvent({
        name: payload.name,
        data: { vertical: getVertical(), ...payload.data },
      });
    } catch (e: any) {
      logger.error({ error: e, tags: [CLIENT_SIDE_ERROR_TAG] });
      /* istanbul ignore next */
      captureException(e);
    }
  };

  /* istanbul ignore next */
  export const setDefaultUserProps = (careDeviceId: string) => {
    try {
      initialize();
      const userProps: any = {};
      const czenSessionId = getCzenSessionId();
      if (czenSessionId) {
        userProps.czen_session_id = czenSessionId;
      }
      const czenVisitorId = getCzenVisitorId();
      if (czenVisitorId) {
        userProps.visitor_id = czenVisitorId;
      }
      userProps.care_device_id = careDeviceId;
      setUserProps({
        props: userProps,
      });
    } catch (e: any) {
      logger.error({ error: e, tags: [CLIENT_SIDE_ERROR_TAG] });
      captureException(e);
    }
  };

  /* istanbul ignore next */
  export const setMemberId = (memberId: string) => {
    try {
      initialize();
      setUserProps({
        memberId: memberId || undefined,
      });
    } catch (e: any) {
      logger.error({ error: e, tags: [CLIENT_SIDE_ERROR_TAG] });
      captureException(e);
    }
  };

  /* istanbul ignore next */
  export const logTestExposure = (testName: string, evaluationDetail: FeatureFlags[string]) => {
    if (!evaluationDetail) return;

    // only log the exposure if the user matched the flag's inclusion rules,
    // in which case we also know that we have a non-null variationIndex since LD requires a variation
    // to be specified for each rule
    const loggableTestExposureReasons = ['RULE_MATCH', 'OVERRIDE'];
    if (loggableTestExposureReasons.includes(evaluationDetail.reason?.kind)) {
      AnalyticsHelper.logEvent({
        name: 'Test Exposure',
        data: {
          test_name: testName,
          test_variant: evaluationDetail.variationIndex!.toString(),
        },
      });
    }
  };

  /* istanbul ignore next */
  export const logScreenViewed = async (screenName: string, source: string) => {
    AnalyticsHelper.logEvent({
      name: 'Screen Viewed',
      data: {
        screen_name: screenName,
        source,
      },
    });
  };
}

export const formatAmplitudeDate = (date: string | null) => {
  if (date == null) return '';
  return dayjs(date).format(AMPLITUDE_DATE_FORMAT);
};

export const formatAmplitudeTime = (time: string) => {
  const [hour, minute] = time.split(':');
  return dayjs().hour(Number(hour)).minute(Number(minute)).format(AMPLITUDE_TIME_FORMAT);
};
