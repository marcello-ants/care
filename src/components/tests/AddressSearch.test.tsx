import { fireEvent, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { preRenderPage } from '@/__setup__/testUtil';
import { AddressInput } from '@/__generated__/globalTypes';
import { AnalyticsHelper } from '@/utilities/analyticsHelper';
import AddressSearch from '../AddressSearch';

jest.mock('@/utilities/analyticsHelper', () => ({
  ...jest.requireActual('@/utilities/analyticsHelper'),
  AnalyticsHelper: {
    logEvent: jest.fn(),
  },
}));

const initialSeekerAddress = {
  addressLine1: '',
  city: '',
  state: '',
  zip: '',
};

const ADDRESS_LINE1_MOCK = 'address';
const DISPLAYED_ADDRESS_MOCK = 'address \n address2, MA 02453';
const ZIPCODE_MOCK = '02453';

const setSeekerAddress = jest.fn();
const setDisplayAddress = jest.fn();

function renderPage(overrideSeekerAddress?: AddressInput, overrideDisplayAddress?: string) {
  const { renderFn, routerMock } = preRenderPage();

  const view = renderFn(
    <AddressSearch
      seekerAddress={overrideSeekerAddress || initialSeekerAddress}
      setSeekerAddress={setSeekerAddress}
      displayAddress={overrideDisplayAddress || ''}
      setDisplayAddress={setDisplayAddress}
      pageName="Test"
    />
  );

  return { view, routerMock };
}

describe('AddressSearch', () => {
  it('Should match a snapshot', () => {
    const { view } = renderPage();

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('Should change the label on input focus and blur', () => {
    renderPage();

    const streetAddressInput = screen.getByRole('textbox', { name: 'Street address' });
    fireEvent.focus(streetAddressInput);

    const searchAddressInput = screen.getByRole('textbox', { name: 'Search address' });

    expect(searchAddressInput).toBeInTheDocument();

    fireEvent.blur(searchAddressInput);

    expect(streetAddressInput).toBeInTheDocument();
  });

  it('Should render the Edit mode with existing address information and switch to Search mode on Edit click', () => {
    const seekerAddress: AddressInput = {
      ...initialSeekerAddress,
      addressLine1: ADDRESS_LINE1_MOCK,
      zip: ZIPCODE_MOCK,
    };

    renderPage(seekerAddress, DISPLAYED_ADDRESS_MOCK);

    const editButton = screen.getByRole('button', { name: 'Edit' });

    expect(editButton).toBeInTheDocument();

    fireEvent.click(editButton);

    const streetAddressInput = screen.getByRole('textbox', { name: 'Street address' });

    expect(streetAddressInput).toBeInTheDocument();
  });

  it('Should render the Edit mode, display error message and log an error event', () => {
    const seekerAddress: AddressInput = {
      ...initialSeekerAddress,
      addressLine1: ADDRESS_LINE1_MOCK,
    };

    renderPage(seekerAddress, DISPLAYED_ADDRESS_MOCK);

    const editButton = screen.getByRole('button', { name: 'Edit' });
    const errorMessageText = 'Invalid address';
    const errorMessage = screen.getByText(errorMessageText);

    expect(editButton).toBeInTheDocument();
    expect(errorMessage).toBeInTheDocument();
    expect(AnalyticsHelper.logEvent).toHaveBeenCalledWith({
      name: 'Error Viewed',
      data: {
        enrollment_step: 'Test',
        error_type: 'Form Validation Error',
        error: errorMessageText,
      },
    });
  });

  it('Should set addressLine 2 on Apt or Suite input change', () => {
    const seekerAddress: AddressInput = {
      ...initialSeekerAddress,
      addressLine1: ADDRESS_LINE1_MOCK,
    };

    renderPage(seekerAddress, DISPLAYED_ADDRESS_MOCK);

    const suiteInput = screen.getByRole('textbox', { name: 'Apt or Suite (optional)' });

    expect(suiteInput).toBeInTheDocument();

    userEvent.type(suiteInput, '2');

    expect(setSeekerAddress).toHaveBeenCalledWith({
      ...seekerAddress,
      addressLine2: '2',
    });
  });
});
