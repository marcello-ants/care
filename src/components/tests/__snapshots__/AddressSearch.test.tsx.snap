// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressSearch Should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-8"
  >
    <div
      class="MuiBox-root MuiBox-root-9 makeStyles-addressSearchContainer-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-root-15 makeStyles-loopIcon-2"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 17.29a7 7 0 117-7 7 7 0 01-7 7zm0-12.63a5.53 5.53 0 105.52 5.52A5.51 5.51 0 0010 4.66z"
          fill="#23405f"
        />
        <path
          d="M20.24 20.77a.83.83 0 01-.53-.22l-5.59-5.37a.75.75 0 111.06-1.06l5.59 5.37a.72.72 0 010 1.06.58.58 0 01-.53.22z"
          fill="#23405f"
        />
      </svg>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-inline-11"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          for="search-address"
          id="search-address-label"
        >
          Street address
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="search-address"
            name="search address"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
